vra/resource_deployment.go

```diff
@@ -380,6 +386,46 @@ func resourceDeploymentRead(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+// Gets the inputs and their types as map[string]string
+func getInputTypesMap(d *schema.ResourceData, apiClient *client.MulticloudIaaS) map[string]string {
+	inputTypesMap := make(map[string]string)
+
+	if _, ok := d.GetOk("inputs"); !ok {
+		return inputTypesMap
+	}
+
+	// Get blueprint_id and catalog_item_id
+	blueprintID, catalogItemID := "", ""
+	if v, ok := d.GetOk("blueprint_id"); ok {
+		blueprintID = v.(string)
+	}
+
+	if v, ok := d.GetOk("catalog_item_id"); ok {
+		catalogItemID = v.(string)
+	}
+
+	if catalogItemID != "" {
+		// Get the catalog item inputs and their types
+		catalogItemVersion := ""
+		if v, ok := d.GetOk("catalog_item_version"); ok {
+			catalogItemVersion = v.(string)
+		}
+
+		inputTypesMap, _ = getCatalogItemInputTypesMap(apiClient, catalogItemID, catalogItemVersion)
+	} else if blueprintID != "" {
+		// Get the blueprint inputs and their types
+		blueprintVersion := ""
+		if v, ok := d.GetOk("blueprint_version"); ok {
+			blueprintVersion = v.(string)
+		}
+
+		inputTypesMap, _ = getBlueprintInputTypesMap(apiClient, blueprintID, blueprintVersion)
+	}
+
```

Comment:
Is it an error if neither blueprint_id or catalog_item_id are specified?

Reply:
A deployment can be requested with one of (blueprint_id, catalog_item_id, blueprint_content). However, the create operation fails if the configuration provided has both `blueprint_content` and `inputs` as the inputs schema cannot be retrieved from vRA, but needs the parsing of the blueprint_content in the provider.

Hence, in the read, if the deployment has blueprint_id or catalog_item_id, their respective inputs schema is retrieved before mapping the values to state. If the deployment is created with inline blueprint and inputs (outside of terraform vra provider), one should still be able to import them.

Update day-2 action (with a different set of inputs from the time of create) is not supported on deployments with an inline blueprint.